core/src/main/java/cz/o2/proxima/util/Classpath.java

```diff
@@ -37,7 +37,7 @@
   public static <T> Class<T> findClass(String name, Class<T> superClass) {
 
     Class clz;
-    if ((clz = findClass(name)) != null) {
+    if ((clz = findClass(name)) != null && superClass.isAssignableFrom(clz)) {
```

Comment:
Can we please check this after the class has been found? I.e.
```
  if ((clz = findClass(name)) != null) {
    Precondition.checkState(superClass.isAssignableFrom(clz), "Class " + clz + " is not " + superClass);
  }
```

Reply:
Fixed